packages/block-editor/src/store/selectors.js

```diff
@@ -1734,6 +1735,40 @@ export const __experimentalGetAllowedBlocks = createSelector(
 	]
 );
 
+/**
+ * Returns the list of allowed patterns for inner blocks children
+ *
+ * @param {Object}  state        Editor state.
+ * @param {?string} rootClientId Optional target root client ID.
+ *
+ * @return {Array?} The list of allowed block types.
+ */
+export const __experimentalGetAllowedPatterns = createSelector(
+	( state, rootClientId = null ) => {
+		const patterns = state.settings.__experimentalBlockPatterns;
+
+		if ( ! rootClientId ) {
+			return patterns;
+		}
+
+		const patternsAllowed = filter( patterns, ( { content } ) => {
+			const blocks = parse( content );
```

Comment:
Parsing is a heavy operation and doing on a selector (which is actually the bottleneck for redux applications) is something we should consider very carefully. Did you notice any impact on  the performance of the patterns tab? Do we have any measures about that tab?

Reply:
So far it's adding up +100ms to the loading of the inserter each time I select an element with a different parent. Local dev installation with TT1B theme. In production, it will be a bit faster but way too much either way, and don't forget my dev machine is probably beefier than the average users'. This is definitely unacceptable if we have more patterns.

Was there any exploration about memoizing/implementing a cache for `parse`?

Reply:
Just pushed a commit which creates a selector for parsed patterns. Now the parsing is only done once. So only the first open of the inserter is affected.